src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1734,6 +1734,14 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode, callable $e
                         ->scalarNode('mock_response_factory')
                             ->info('The id of the service that should generate mock responses. It should be either an invokable or an iterable.')
                         ->end()
+                        ->arrayNode('uri_template')
+                            ->canBeEnabled()
+                            ->children()
+                                ->scalarNode('expander')
```

Comment:
I'm not sold on the need for this.
Instead, I'd define the expander as a service with a stable id and I'll let ppl redefine the service if needed
about the config, the enable/disable toggle is nice. what about supporting default vars also (and instead of expander)?